libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/roadobject/merge/MergingArea.kt

```diff
@@ -0,0 +1,60 @@
+package com.mapbox.navigation.base.trip.model.roadobject.merge
+
+import com.mapbox.navigation.base.trip.model.roadobject.RoadObject
+import com.mapbox.navigation.base.trip.model.roadobject.RoadObjectProvider
+import com.mapbox.navigation.base.trip.model.roadobject.RoadObjectType
+
+/**
+ * Road object type that provides information about merging areas on the road.
+ *
+ * @param info Merging Area specific details, see [MergingAreaInfo].
+ * @see RoadObject
+ * @see RoadObjectType.MERGING_AREA
+ */
+internal class MergingArea internal constructor(
```

Comment:
I made this all internal, because I want to first merge this PR and then support MergingAreas properly, so please don't focus on this implementation at the moment.